src/clear/sql/query/with_pagination.cr

```diff
@@ -1,9 +1,26 @@
 module Clear::SQL::Query::WithPagination
+  DEFAULT_LIMIT = 50
+  DEFAULT_PAGE = 1
   
   macro included
     property total_entries : Int64? = nil  
   end
 
+  # Maybe this goes on the Collection?
+  def paginate(page : Int32 = DEFAULT_PAGE, per_page : Int32 = DEFAULT_LIMIT)
+    # Need to clear these values to get total count first
+    clear_limit.clear_offset
+    # TODO: this fails. Maybe something with clear_select? 
+    # @total_entries = count
```

Comment:
@anykeyh I ran in to some issues here that I need some help with. Calling `count` here causes my specs to fail:

```
  1) Clear::Model with pagination can pull the next 5 users from page 2
     Failure/Error: users.map(&.first_name).should eq ["user5", "user6", "user7", "user8", "user9"]

       Expected: ["user5", "user6", "user7", "user8", "user9"]
            got: []

     # spec/model/model_spec.cr:342
```

If I raised `users.to_sql`, it would return `SELECT COUNT(*) FROM model_users LIMIT 5 OFFSET 5`. The limit and offset were correct, but I wasn't sure where that count was coming from. If I tried to raise `users.first`, I'd get a PG error complaining about `model_users.id` not being in the GROUP BY clause. I added the below "hack" to get around that, but obviously there needs to be a better solution.

Reply:
I don't have so much time during the next days, I'll test your branch once I have some time 👍

Reply:
Cool, thanks!